cloud/google/machineactuator.go

```diff
@@ -345,13 +350,19 @@ func (gce *GCEClient) Delete(machine *clusterv1.Machine) error {
 		return nil
 	}
 
-	config, err := gce.providerconfig(machine.Spec.ProviderConfig)
+	machineConfig, err := gce.machineproviderconfig(machine.Spec.ProviderConfig)
```

Comment:
I think it would help the user if the the error said which providerconfig (ie. the cluster's or the machine's) it could not unmarshal.

Reply:
Fixed